Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/ReferalPayout.hs

```diff
@@ -0,0 +1,166 @@
+import Data.OpenApi (ToSchema)
+import Data.Text hiding (map)
+import Data.Time.Calendar
+import qualified Domain.Action.UI.Driver as DD
+import qualified Domain.Action.UI.Payout as DAP
+import qualified Domain.Types.DailyStats as DS
+import qualified Domain.Types.DriverFee as DFee
+import qualified Domain.Types.Extra.MerchantServiceConfig as DEMSC
+import Domain.Types.Merchant
+import Domain.Types.MerchantOperatingCity
+import Domain.Types.Person
+import qualified Domain.Types.Plan as DPlan
+import qualified Environment
+import EulerHS.Prelude hiding (id)
+import Kernel.Beam.Functions
+import qualified Kernel.External.Payment.Juspay.Types.Common as JuspayT
+import qualified Kernel.External.Payment.Juspay.Types.CreateOrder
+import qualified Kernel.External.Payout.Interface.Types as Payout
+import qualified Kernel.External.Payout.Types as PT
+import Kernel.External.Types (ServiceFlow)
+import qualified Kernel.Prelude
+import qualified Kernel.Storage.Esqueleto as Esq
+import qualified Kernel.Storage.Hedis as Redis
+import qualified Kernel.Types.APISuccess
+import qualified Kernel.Types.Error
+import qualified Kernel.Types.Id
+import qualified Kernel.Types.Price (Currency (..))
+import Kernel.Utils.Common
+import qualified Lib.Payment.Domain.Action as Payout
+import qualified Lib.Payment.Domain.Types.Common as DLP
+import qualified Lib.Payment.Storage.Queries.PaymentOrder as QOrder
+import qualified Lib.Payment.Storage.Queries.PayoutOrders as QPayoutOrder
+import Servant hiding (throwError)
+import qualified Storage.CachedQueries.Merchant.MerchantOperatingCity as CQMOC
+import qualified Storage.CachedQueries.Merchant.PayoutConfig as CPC
+import qualified Storage.Queries.DailyStats as QDS
+import qualified Storage.Queries.DriverInformation as DrInfo
+import qualified Storage.Queries.DriverStats as QDriverStats
+import qualified Storage.Queries.Person as PersonQuery
+import qualified Storage.Queries.Vehicle as QVeh
+import Tools.Auth
+import Tools.Error
+import qualified Tools.Payout as TP
+
+getReferralEarnings ::
+  ( ( Kernel.Prelude.Maybe (Kernel.Types.Id.Id Domain.Types.Person.Person),
+      Kernel.Types.Id.Id Domain.Types.Merchant.Merchant,
+      Kernel.Types.Id.Id Domain.Types.MerchantOperatingCity.MerchantOperatingCity
+    ) ->
+    Data.Time.Calendar.Day ->
+    Data.Time.Calendar.Day ->
+    Environment.Flow API.Types.UI.ReferalPayout.ReferralEarningsRes
+  )
+getReferralEarnings (mbPersonId, _merchantId, merchantOpCityId) fromDate toDate = do
+  personId <- mbPersonId & fromMaybeM (PersonNotFound "No person found")
+  let dates = [fromDate .. toDate]
+  earnings <- catMaybes <$> (forM dates $ \date -> QDS.findByDriverIdAndDate personId date)
+  driverStats <- runInReplica $ QDriverStats.findById personId >>= fromMaybeM (PersonNotFound personId.getId)
+  dInfo <- runInReplica $ DrInfo.findByPrimaryKey personId >>= fromMaybeM DriverInfoNotFound
+  vehicle <- QVeh.findById personId >>= fromMaybeM (DriverWithoutVehicle personId.getId)
```

Comment:
What if driver had vehicle but later he removed his vehicle? Why should we break this API?